src/tests/ftest/container/rebuild_tests.py

```diff
@@ -0,0 +1,175 @@
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+import json
+import os
+import string
+import random
+import sys
+
+from avocado import Test
+
+sys.path.append('./util')
+sys.path.append('../util')
+sys.path.append('../../../utils/py')
+sys.path.append('./../../utils/py')
+
+import write_host_file
+import server_utils
+import AgentUtils
+from daos_api import (DaosContext, DaosLog, DaosApiError, DaosServer, DaosPool,
+                      DaosContainer)
+
+class RebuildContainerCreate(Test):
+    """
+    Tests container creation while a rebuild is in progress.
+    """
+
+    def setUp(self):
+        self.hostlist = None
+
+        with open('../../../.build_vars.json') as build_file:
+            self.build_paths = json.load(build_file)
+
+        # setup the DAOS python API + client logging
+        self.context = DaosContext(self.build_paths['PREFIX'] + '/lib/')
+        self.d_log = DaosLog(self.context)
+
+        # init params for pool create
+        self.createuid = os.geteuid()
+        self.creategid = os.getegid()
+        self.createmode = self.params.get("mode",
+                                          '/run/testparams/createmode/')
+        self.createsetid = self.params.get("setname",
+                                           '/run/testparams/createset/')
+        self.createsize = self.params.get("size",
+                                          '/run/testparams/createsize/')
+
+        # how many objects and records are we creating
+        self.objcount = self.params.get("objcount",
+                                        '/run/testparams/numobjects/')
+        self.reccount = self.params.get("reccount",
+                                        '/run/testparams/numrecords/')
+        if self.objcount == 0:
+            self.reccount = 0
+
+        # which rank to write to and kill
+        self.rank = self.params.get("rank", '/run/testparams/ranks/*')
+
+        # how much data to write with each key
+        self.size = self.params.get("size", '/run/testparams/datasize/')
+
+        #Start the server + agent
+        self.server_group = self.params.get("server_group", '/server/',
+                                            'daos_server')
+        self.basepath = os.path.normpath(self.build_paths['PREFIX'] + "/../")
+        self.hostlist = self.params.get("test_machines", '/run/hosts/')
+        self.hostfile = write_host_file.write_host_file(self.hostlist,
+                                                        self.workdir)
+
+        self.agent_sessions = AgentUtils.run_agent(self.basepath, self.hostlist)
+        server_utils.run_server(self.hostfile, self.server_group, self.basepath)
+
+    def tearDown(self):
+        if self.agent_sessions:
```

Comment:
try finally

Reply:
Wrapped sessions kill in try, server kill in finally.